Source/AssetRipper.Processing/AnimatorControllers/AnimatorStateMachineContext.cs

```diff
@@ -0,0 +1,754 @@
+						StateMachines[j] = stateMachine;
+						// set Child StateMachines
+						mainStateMachine.ChildStateMachineP.Add(stateMachine);
+					}
+					else
+					{
+						j--;
+					}
+				}
+
+				// set Child States with their Transitions
+				for (int i = 0; i < StateMachines.Length; i++)
+				{
+					int childStateCount = StateContext.StateIdxsForStateMachine(StateMachines[i].Name).Count();
+					StateMachines[i].SetChildStateCapacity(childStateCount);
+					foreach (int stateIdx in StateContext.StateIdxsForStateMachine(StateMachines[i].Name))
+					{
+						AddStateAndTransitionsToStateMachine(i, stateIdx);
+					}
+				}
+			}
+		}
+	}
+
+	private void AddStateAndTransitionsToStateMachine(int parentStateMachineIdx, int stateIdx)
+	{
+		// -- add child State to stateMachine --
+		IAnimatorState state = StateContext.GetState(stateIdx);
+		IStateConstant stateConstant = StateContext.GetStateConstant(stateIdx);
+		IAnimatorStateMachine parentStateMachine = StateMachines[parentStateMachineIdx];
+		if (parentStateMachine.Has_ChildStates())
+		{
+			ChildAnimatorState childState = parentStateMachine.ChildStates.AddNew();
+			// set childState.Position later, when having all Children set
+			childState.State.SetAsset(parentStateMachine.Collection, state);
+		}
+		else
+		{
+			parentStateMachine.StatesP.Add(state);
+		}
+		// set state.Position later, when having all Children set
+
+		// -- add State Transitions --
+		AssetList<PPtr_AnimatorStateTransition_4>? transitionList = null;
+		if (state.Has_Transitions())
+		{
+			state.Transitions.Capacity = stateConstant.TransitionConstantArray.Count;
+		}
+		else if (parentStateMachine.Has_OrderedTransitions())
+		{
+			//I'm not sure if this is correct, but it seems to be the only logical way to store the transitions before Unity 5.
+			//IAnimatorStateMachine.LocalTransitions only exists until Unity 4.2.0, so by process of elimination, this is the only option.
+
+			AssetPair<PPtr_AnimatorState_4, AssetList<PPtr_AnimatorStateTransition_4>> pair = parentStateMachine.OrderedTransitions.AddNew();
+			pair.Key.SetAsset(parentStateMachine.Collection, state);
+			transitionList = pair.Value;
+		}
+		for (int j = 0; j < stateConstant.TransitionConstantArray.Count; j++)
+		{
+			ITransitionConstant transitionConstant = stateConstant.TransitionConstantArray[j].Data;
+			IAnimatorStateTransition? transition = CreateAnimatorStateTransition(parentStateMachineIdx, transitionConstant);
+			if (transition != null)
+			{
+				if (state.Has_Transitions())
+				{
+					state.TransitionsP.Add(transition);
+				}
+				else
+				{
+					transitionList?.AddNew().SetAsset(parentStateMachine.Collection, transition);
+				}
+			}
+		}
+	}
+
+	private IAnimatorStateTransition? CreateAnimatorStateTransition(int parentStateMachineIdx, ITransitionConstant Transition)
+	{
+		if (!TryGetDestinationState(Transition.DestinationState, out IAnimatorState? stateDst, out int stateMachineDstIdx, out bool isEntryDst))
```

Comment:
Please don't use abbreviations. `stateDestination`, `stateMachineDestinationIndex`, `isEntryDestination`. I know Unity does it, but I prefer names that are more readable. After your pull request is merged, I want to rename the Unity fields to align with my naming standards.

https://github.com/AssetRipper/AssemblyDumper/commit/111228a279deab1a7f38df84a2ede4431d247e7c